src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionsController.kt

```diff
@@ -94,20 +97,40 @@ class IdentificationSessionsController(
             }
             .defaultIfEmpty(ResponseEntity.status(HttpStatus.NOT_FOUND).body(null))
             .doOnError { exception ->
-                log.error { "error occurred while getting the tc token for session with id $useIDSessionId; ${exception.message}" }
+                log.error { "error occurred while getting tc token for useIDSessionId $useIDSessionId;\n ${exception.message}" }
             }
             .onErrorReturn(
                 ResponseEntity.internalServerError().body(null)
             )
     }
 
     @GetMapping("/{eIDSessionId}")
-    fun getIdentity(@PathVariable eIDSessionId: UUID): Mono<IdentityAttributes> {
-        // Currently, mock identity
-        return Mono.just(IdentityAttributes("firstname", "lastname"))
-            .filter {
-                identificationSessionService.sessionExists(eIDSessionId)
+    fun getIdentity(@PathVariable eIDSessionId: UUID): Mono<ResponseEntity<GetResultResponseType>> {
+        return identificationSessionService.findByEIDSessionId(eIDSessionId)
+            .map {
+                eidService.getEidInformation(eIDSessionId.toString())
+            }
+            .doOnNext {
+                // TODO: REPLACED CONTAINS CONDITION WITH FIXED SUCCESS CASE (e.g. http://www.bsi.bund.de/ecard/api/1.1/resultmajor#success) AS SOON AS WE HAVE E2E SUCCESS FLOW DONE
+                if (it.result.resultMajor.contains("api")) {
```

Comment:
why do we need to check the value of the result? did we decide to handle the error in our backend or let the eService do error handling?

Reply:
We gotta decide on this one. 
Merlin and I said that it would be nice have different error cases to handle. 
Also, we only want to remove the session if the result was successfully given back; so we need at least a success case.